contracts/mainnet/connectors/swap-aggregator/main.sol

```diff
@@ -0,0 +1,95 @@
+//SPDX-License-Identifier: MIT
+pragma solidity 0.8.19;
+
+import "./helpers.sol";
+import "./events.sol";
+
+contract SwapAggregatorConnector is Helpers, Events {
+    struct RouterInfo {
+        address router;
+        address allowanceHolder;
+    }
+
+    mapping(string => RouterInfo) public routers;
+
+    address public chief;
+    address public master;
+
+    modifier onlyChiefOrMaster() {
+        require(msg.sender == chief || msg.sender == master, "Not authorized");
+        _;
+    }
+
+    modifier onlyMaster() {
+        require(msg.sender == master, "Not authorized");
+        _;
+    }
+
+    constructor(address _chief, address _master) {
+        chief = _chief;
+        master = _master;
+    }
+
+    function updateChief(address _newChief) external onlyMaster {
+        require(_newChief != address(0), "Invalid chief address");
+        chief = _newChief;
+    }
+
+    function setRouter(
+        string calldata connectorName,
+        address router,
+        address allowanceHolder
+    ) external onlyChiefOrMaster {
+        routers[connectorName] = RouterInfo(router, allowanceHolder);
+    }
+
+    function swap(
+        address buyAddr,
+        address sellAddr,
+        uint256 sellAmt,
+        uint256 unitAmt,
+        bytes calldata callData,
+        string calldata connectorName,
+        uint256 setId
+    )
+        external
+        payable
+        returns (string memory _eventName, bytes memory _eventParam)
+    {
+        require(
+            routers[connectorName].router != address(0),
+            "Connector not enabled"
+        );
+        RouterInfo memory info = routers[connectorName];
```

Comment:
@SamarendraGouda This won't work. You need to call to this contract as a view function,